plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -170,15 +170,15 @@ public void setPropertyValue(String path, Object value) {
     }
 
     public String getIndex() {
```

Comment:
it is not lenient if you throw exception when you don't find something :)  I somehow thought that strings would be ok here, to have more flexibility and less dependencies on core. But maybe it is me being paranoid. We can probably start with your approach and see if it's ok.